clang/include/clang/AST/Type.h

```diff
@@ -1996,7 +1996,7 @@ class alignas(TypeAlignment) Type : public ExtQualsTypeCommonBase {
     TypeBits.Dependence = static_cast<unsigned>(Dependence);
     TypeBits.CacheValid = false;
     TypeBits.CachedLocalOrUnnamed = false;
-    TypeBits.CachedLinkage = NoLinkage;
+    TypeBits.CachedLinkage = llvm::to_underlying(Linkage::Invalid);
```

Comment:
CC @AaronBallman This passes Clang tests with all those `llvm_unreachable` around, so I'm rather confident about this. Note that I removed `NFC` tag from the title.